timely/src/dataflow/operators/core/partition.rs

```diff
@@ -0,0 +1,75 @@
+//! Partition a stream of records into multiple streams.
+
+use timely_container::{Container, ContainerBuilder, PushInto, SizableContainer};
+
+use crate::dataflow::channels::pact::Pipeline;
+use crate::dataflow::operators::generic::builder_rc::OperatorBuilder;
+use crate::dataflow::{Scope, StreamCore};
+use crate::Data;
+
+/// Partition a stream of records into multiple streams.
+pub trait Partition<G: Scope, C: Container> {
+    /// Produces `parts` output streams, containing records produced and assigned by `route`.
+    ///
+    /// # Examples
+    /// ```
+    /// use timely::dataflow::operators::ToStream;
+    /// use timely::dataflow::operators::core::{Partition, Inspect};
+    ///
+    /// timely::example(|scope| {
+    ///     let streams = (0..10).to_stream(scope)
+    ///                          .partition(3, |x| (x % 3, x));
+    ///
+    ///     for (idx, stream) in streams.into_iter().enumerate() {
+    ///         stream
+    ///             .container::<Vec<_>>()
+    ///             .inspect(move |x| println!("seen {idx}: {x:?}"));
+    ///     }
+    /// });
+    /// ```
+    fn partition<CB, D2, F>(&self, parts: u64, route: F) -> Vec<StreamCore<G, CB::Container>>
+    where
+        CB: ContainerBuilder,
+        CB::Container: SizableContainer + PushInto<D2> + Data,
```

Comment:
The `SizableContainer` requirement here is not needed, right? That's a requirement that'll be imposed by the container builder, but the implementation should only require the remaining traits for the container.

Reply:
Fixed in #612.